lib/generate-plugin/plugin-generator.js

```diff
@@ -0,0 +1,89 @@
+const path = require('path');
+const mkdirp = require('mkdirp');
+const _ = require('lodash');
+const Generator = require('yeoman-generator');
+
+/**
+ * A yeoman generator class for creating a webpack
+ * plugin project. It adds some starter plugin code
+ * and runs `webpack-defaults`.
+ *
+ * @class PluginGenerator
+ * @extends {Generator}
+ */
+class PluginGenerator extends Generator {
+	prompting() {
+		const prompts = [
+			{
+				type: 'input',
+				name: 'name',
+				message: 'Plugin name',
+				default: 'my-webpack-plugin',
+				filter: _.kebabCase,
+				validate: str => str.length > 0,
+			},
+		];
+
+		return this.prompt(prompts).then(props => {
+			this.props = props;
+		});
+	}
+
+	default() {
+		if (path.basename(this.destinationPath()) !== this.props.name) {
+			this.log(
+				'Your plugin must be inside a folder named ' +
+				this.props.name +
+				'\nI will create this folder for you.'
+			);
+			mkdirp(this.props.name);
+			this.destinationRoot(this.destinationPath(this.props.name));
+		}
+	}
+
+	writing() {
+		const copy = (filePath) => {
```

Comment:
Maybe abstract this into a util file so you'd avoid re-declaring the same function twice, if it contains the same logic